src/app/loadout-builder/LoadoutBuilderContainer.tsx

```diff
@@ -0,0 +1,345 @@
+  }
+}
+
+function mapStateToProps() {
+  const itemsSelector = createSelector(
+    storesSelector,
+    (
+      stores
+    ): Readonly<{
+      [classType: number]: ItemsByBucket;
+    }> => {
+      const items: {
+        [classType: number]: { [bucketHash: number]: DimItem[] };
+      } = {};
+      for (const store of stores) {
+        for (const item of store.items) {
+          if (!item || !item.isDestiny2() || !isLoadoutBuilderItem(item)) {
+            continue;
+          }
+          for (const classType of item.classType === DestinyClass.Unknown
+            ? [DestinyClass.Hunter, DestinyClass.Titan, DestinyClass.Warlock]
+            : [item.classType]) {
+            if (!items[classType]) {
+              items[classType] = {};
+            }
+            if (!items[classType][item.bucket.hash]) {
+              items[classType][item.bucket.hash] = [];
+            }
+            items[classType][item.bucket.hash].push(item);
+          }
+        }
+      }
+
+      return items;
+    }
+  );
+
+  return (state: RootState): StoreProps => ({
+    storesLoaded: storesLoadedSelector(state),
+    stores: sortedStoresSelector(state),
+    isPhonePortrait: state.shell.isPhonePortrait,
+    items: itemsSelector(state),
+    defs: state.manifest.d2Manifest,
+    searchConfig: searchConfigSelector(state),
+    filters: searchFiltersConfigSelector(state),
+  });
+}
+
+/**
+ * The Loadout Optimizer screen
+ */
+function LoadoutBuilderContainer({
+  account,
+  stores,
+  isPhonePortrait,
+  items,
+  defs,
+  searchConfig,
+  filters,
+  location,
+}: Props) {
+  const [
+    {
+      lockedMap,
+      lockedSeasonalMods,
+      lockedArmor2Mods,
+      selectedStoreId,
+      statFilters,
+      minimumPower,
+      query,
+      statOrder,
+      assumeMasterwork,
+    },
+    stateDispatch,
+  ] = useReducer(stateReducer, { stores, location }, init);
+
+  useSubscription(
+    useCallback(
+      () =>
+        D2StoresService.getStoresStream(account).subscribe((stores) => {
+          if (!stores || !stores.length) {
+            return;
+          }
+
+          if (!selectedStoreId) {
+            stateDispatch({ type: 'changeCharacter', storeId: getCurrentStore(stores)!.id });
+          }
+        }),
+      /* do not include selectedStoreId in dependencies, it triggers two process rounds after changing store */
+      /* eslint-disable react-hooks/exhaustive-deps */
+      [account]
+    )
+  );
+
+  useSubscription(
+    useCallback(
+      () => refresh$.subscribe(() => queueAction(() => D2StoresService.reloadStores())),
+      []
+    )
+  );
+
+  const enabledStats = useMemo(
+    () => new Set(statKeys.filter((statType) => !statFilters[statType].ignored)),
+    [statFilters]
+  );
+
+  const store = stores.find((s) => s.id === selectedStoreId)!;
+
+  if (!store || !defs) {
+    return <ShowPageLoading message={t('Loading.Profile')} />;
+  }
+
+  return (
+    <LoadoutBuilder
```

Comment:
I don't actually think this needs to be a container. I just need to ensure stores and defs are loaded and if not we render the DIM loading page. I see the split as necessary as I dont want to run the hooks currently in loadout builder unless stores and defs are available and this kind of an approximation of a return early condition, just wrapped in a component.

I would be open to other options to do this and in fact I would like to move all state that isn't stores or defs back down into the LoadoutBuilder. This would also solve the passing all the props issue and I think it would help the issue I am having with the LO config PR as well.